src/components/network-map-viewer/network/network-map.jsx

```diff
@@ -523,6 +528,118 @@ const NetworkMap = (props) => {
         setCentered(INITIAL_CENTERED);
     }, [mapLib?.key]);
 
+    const [features, setFeatures] = useState({});
+
+    useEffect(() => {
+        props.onFeaturesChanged(features);
+    }, [features]);
+
+    const onUpdate = useCallback((e) => {
+        setFeatures((currFeatures) => {
+            const newFeatures = { ...currFeatures };
+            for (const f of e.features) {
+                newFeatures[f.id] = f;
+            }
+            return newFeatures;
+        });
+    }, []);
+
+    const getPolygonFeatures = () => features;
+    const getSelectedSubstations = () => {
+        const substations = getSubstationsInPolygon(
+            features,
+            props.mapEquipments,
+            props.geoData
+        );
+        return (
+            substations.filter((substation) => {
+                return substation.voltageLevels.some((vl) =>
+                    props.filteredNominalVoltages.includes(vl.nominalV)
+                );
+            }) ?? []
+        );
+    };
+
+    const getSelectedVoltageLevel = () => {
+        const selectedVL = getVoltageLevelFromSubstation(
+            getSelectedSubstations()
+        );
+        return selectedVL.filter((vl) =>
+            props.filteredNominalVoltages.includes(vl.nominalV)
+        );
+    };
+
+    function getSelectedLinesInPolygon(
```

Comment:
move `getSelectedLinesInPolygon` outside the component scope as you did for `getSubstationsInPolygon` ?